mempool/clist_mempool.go

```diff
@@ -572,13 +642,17 @@ func (mem *CListMempool) ReapMaxTxs(max int) types.Txs {
 	defer mem.updateMtx.RUnlock()
 
 	if max < 0 {
-		max = mem.txs.Len()
+		max = mem.Size()
 	}
 
-	txs := make([]types.Tx, 0, cmtmath.MinInt(mem.txs.Len(), max))
-	for e := mem.txs.Front(); e != nil && len(txs) <= max; e = e.Next() {
-		memTx := e.Value.(*mempoolTx)
-		txs = append(txs, memTx.tx)
+	txs := make([]types.Tx, 0, cmtmath.MinInt(mem.Size(), max))
+	iter := NewWRRIterator(mem)
+	for len(txs) <= max {
```

Comment:
I know this isn't introduced in this PR, but this doesn't respect the `max` limit, it may return one more transaction than those that fit in `max`